lib/spack/spack/filesystem_view.py

```diff
@@ -73,16 +73,13 @@ def view_copy(src, dst, view, spec=None):
         # will have the old sbang location in their shebangs.
         # TODO: Not sure which one to use...
         import spack.hooks.sbang as sbang
+        # import here to avoid circular import error
+        import spack.util.file_permissions as fp
         orig_sbang = '#!/bin/bash {0}/bin/sbang'.format(spack.paths.spack_root)
         new_sbang = sbang.sbang_shebang_line()
 
         prefix_to_projection = OrderedDict({
-            spec.prefix: view.get_projection_for_spec(spec),
-            spack.paths.spack_root: view._root})
-
-        for dep in spec.traverse():
-            prefix_to_projection[dep.prefix] = \
-                view.get_projection_for_spec(dep)
```

Comment:
I think we still need the dependency prefixes in the translation dict

Reply:
Okay I've updated this to be the same, just excluding externals from the `prefix_to_projection`